reference-architecture/aws-ansible/playbooks/roles/etcd-volume/tasks/main.yaml

```diff
@@ -0,0 +1,25 @@
+---
+- name: Create filesystem for /var/lib/etcd
+  filesystem:
+      fstype: xfs
+      dev: /dev/xvdc
+
+- name: Create local volumes directory
+  file:
+      path: /var/lib/etcd
+      state: directory
+      recurse: yes
+
+- name: Create fstab entry
+  mount:
+      name: /var/lib/etcd
+      src: '/dev/xvdc'
+      fstype: xfs
+      state: present
+
+- name: Mount fstab entry
+  mount:
+      name: /var/lib/etcd
+      src: '/dev/xvdc'
+      fstype: xfs
+      state: mounted
```

Comment:
VMWare is definitely the outlier...  Of course, GCP and Azure don't natively support cloud-init, but they do have alternatives for doing the same type of actions.

For VM-based deployments, it might be better to do something one-off just for those, or possibly not even support a separate etcd directory by default (unless they provide an image that has it pre-baked).

That said, we could require a VM image that uses LVM and has all the LVs created for these types of things out of the box.